context.tags/addon.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2017 zosky
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import xbmc
+import xbmcgui
+import xbmcaddon
+import json
+
+
+def main():
+    # GET ITEM DATA
+    assetID = int(xbmc.getInfoLabel('ListItem.DBID'))
+    assetTYPE = xbmc.getInfoLabel('ListItem.DBtype')
+
+    # GET ITEM TAGS
+    GETassetTAGS = {'jsonrpc': '2.0', 'id': 'tagSelector',
+                    'method': 'VideoLibrary.Get' + assetTYPE + 'Details',
+                    'params': {assetTYPE + 'id': assetID,
+                               'properties': ['tag']}}
+
+    assetTAGS = json.loads(xbmc.executeJSONRPC(json.dumps(GETassetTAGS)))
+    assetTAGS = assetTAGS['result'][assetTYPE + 'details']['tag']
+
+    # GET ALL TAGS (FOR assetTYPE)
+    GETallTAGS = {'jsonrpc': '2.0', 'id': 'tagSelector',
+                  'method': 'VideoLibrary.GetTags',
+                  'params': {'type': assetTYPE}}
+
+    allTAGS = json.loads(xbmc.executeJSONRPC(json.dumps(GETallTAGS)))
+    allTAGS = allTAGS['result']['tags']
+
+    # MASSAGE TAG LIST INTO MULTI-SELECT FORMAT
+    tags = []
+    for item in allTAGS:
+        tags.append(item['label'])
+
+    # IDENTIFY PRESELECTED TAGS
+    preSelectedTAGS = []
+    for item in assetTAGS:
+        preSelectedTAGS.append(tags.index(item))
+
+    # MAKE DIALOG
+    dialog = xbmcgui.Dialog()
+    title = xbmcaddon.Addon().getLocalizedString(32001)
+    returned = dialog.multiselect(title, tags, preselect=preSelectedTAGS)
+    # IF NOT CANCELED
+    if returned is not None:
+        selected = []
+        for n in returned:
+            selected.append(tags[n])
+
+        # SUBMIT
+        SETnewTAGS = {'jsonrpc': '2.0', 'id': 'tagSelector',
```

Comment:
i'm not explicitly telling it to remove tags that are deselected... it just happens as part of how kodi handles this method ? 

check this, i started with a movie that had 1 tag 
- deselect it ( send _tag: [ ]_ )
```
02:54:35.198 T:140539475240704   DEBUG: Mysql execute: DELETE FROM tag_link WHERE media_id=1633 AND media_type='movie'
```

- select 3 ( send _tag: ["me", "wife", "kid"]_
```
02:56:09.509 T:140541695956736   DEBUG: Mysql execute: DELETE FROM tag_link WHERE media_id=1633 AND media_type='movie'
02:56:09.514 T:140541695956736   DEBUG: Mysql execute: INSERT INTO tag_link (tag_id,media_id,media_type) VALUES(1,1633,'movie')
02:56:09.536 T:140541695956736   DEBUG: Mysql execute: INSERT INTO tag_link (tag_id,media_id,media_type) VALUES(2,1633,'movie')
02:56:09.540 T:140541695956736   DEBUG: Mysql execute: INSERT INTO tag_link (tag_id,media_id,media_type) VALUES(4,1633,'movie')
```

- deselect tag 2 & 4 ( send _tag: ["me"]_)
```
02:56:57.668 T:140540173838080   DEBUG: Mysql execute: DELETE FROM tag_link WHERE media_id=1633 AND media_type='movie'
02:56:57.669 T:140540173838080   DEBUG: Mysql execute: INSERT INTO tag_link (tag_id,media_id,media_type) VALUES(1,1633,'movie')
```